src/cider/nrepl/middleware/util/nrep.clj

```diff
@@ -0,0 +1,26 @@
+(ns cider.nrepl.middleware.util.nrepl
+  "Common utilities for interaction with the client."
+  (:require
+   [clojure.tools.nrepl.middleware.interruptible-eval :refer [*msg*]]
+   [clojure.tools.nrepl.misc :refer [response-for]]
+   [clojure.tools.nrepl.transport :as transport]))
+
+
+(defn send-user-message
```

Comment:
It could be better indeed. I separated them because on emacs side `display-warning` is a very different mechanism than simple `message`, but I doublt anyone would want to use that for  nrep warnings.  So yes, let me change that.